Mac link error: ld: unknown option: --version-script

Line 16 of the `paddle/fluid/inference/CMakeLists.txt`:
https://github.com/PaddlePaddle/Paddle/blob/898e0ffa210551149cc411a3be366e8d36aeb67b/paddle/fluid/inference/CMakeLists.txt#L16

However, `ld` of MacOS doesn't support `--version-script`, so this line will lead to a link error on Mac:
```
ld: unknown option: --version-script
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [paddle/fluid/inference/libpaddle_fluid.dylib] Error 1
make[1]: *** [paddle/fluid/inference/CMakeFiles/paddle_fluid_shared.dir/all] Error 2
make: *** [all] Error 2
```

--- a/paddle/fluid/inference/CMakeLists.txt
+++ b/paddle/fluid/inference/CMakeLists.txt
@@ -13,8 +13,11 @@ cc_library(paddle_fluid_shared SHARED
     SRCS io.cc
     DEPS ARCHIVE_START ${GLOB_OP_LIB} ${FLUID_CORE_MODULES} ARCHIVE_END)
 set_target_properties(paddle_fluid_shared PROPERTIES OUTPUT_NAME paddle_fluid)
-set(LINK_FLAGS "-Wl,--version-script ${CMAKE_CURRENT_SOURCE_DIR}/paddle_fluid.map")
-set_target_properties(paddle_fluid_shared PROPERTIES LINK_FLAGS "${LINK_FLAGS}")
+if(NOT APPLE)
+  # TODO(liuyiqun): Temporarily disable the link flag because it is not support on Mac.
+  set(LINK_FLAGS "-Wl,--version-script ${CMAKE_CURRENT_SOURCE_DIR}/paddle_fluid.map")
+  set_target_properties(paddle_fluid_shared PROPERTIES LINK_FLAGS "${LINK_FLAGS}")
+endif()
 
 if(WITH_TESTING)
   add_subdirectory(tests/book)

